pkg/agent/sidecar/service/observer/hook.go

```diff
@@ -0,0 +1,38 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package observer provides storage observer
+package observer
+
+import (
+	"context"
+	"time"
+
+	"github.com/vdaas/vald/pkg/agent/sidecar/service/storage"
+)
+
+type BackupInfo struct {
+	StartTime time.Time
+	EndTime   time.Time
+	Bytes     int64
+
+	*storage.StorageInfo
+}
+
+type Hook interface {
+	BeforeProcess(ctx context.Context, info *BackupInfo) (context.Context, error)
+	AfterProcess(ctx context.Context, info *BackupInfo) error
```

Comment:
How about naming as `PreProcess` & `PostProcess`

Reply:
Both is okay, but i think it is better to use the current naming.
it is come from go-redis's Hook interface, it is already used in metrics/db/kvs.
https://github.com/vdaas/vald/blob/782e84b7a4109d4cf4b780d1f71c6a919074e573/internal/observability/metrics/db/kvs/redis/redis.go#L152

Also, the word "preprocess" sounds like modifying the incoming data. but here, not modifying just exporting metadata.

Reply:
Both is okay too. But I don't think pre & post always modifying the data.
but I didn't know about go-redis naming, it's okay this time.